File: node/speed-loop.js
/* JavaScript looped 100000000 times in 0.135 seconds in node v16.13.2 
 * JavaScript looped 100000000 times in 0.112 seconds in node v20.12.2
 * JavaScript looped 100000000 times in 0.036 seconds in node v20.13.1
 */
function main() {
  const cycles = 100000000;
  let start = Date.now();
  for (let i = 0; i < cycles; i++) {
	/* Empty loop */
	}
  let end = Date.now();
  let duration = (end - start) / 1000;
  console.log(" looped %d times in %d seconds", cycles, duration);
  }
main();

File: c/speed-loop.c
# C looped 1000000000 times in 2.576541 second on osx
#include <stdio.h>
#include <time.h>
int main () {
  int cycles = 1000000000;
  clock_t start,end;
  double duration;
  start = clock();
  for (int i = 0 ; i < cycles ; i++) {
	/* Empty Loop */ 
        }
  end = clock();
  duration = ((double)(end -start)) / CLOCKS_PER_SEC;
  printf("C looped %d times in %lf seconds\n", cycles, duration);
 return 0;
}   
